cpp/ql/test/library-tests/ir/ir/raw_ir.expected

```diff
@@ -3224,27 +3254,31 @@ ir.cpp:
 
 #  745| Base& Base::operator=(Base const&)
 #  745|   Block 0
-#  745|     v0_0(void)           = EnterFunction              : 
-#  745|     mu0_1(unknown)       = AliasedDefinition          : 
-#  745|     mu0_2(unknown)       = UnmodeledDefinition        : 
-#  745|     r0_3(glval<Base>)    = InitializeThis             : 
-#-----|     r0_4(glval<Base &>)  = VariableAddress[p#0]       : 
-#-----|     mu0_5(Base &)        = InitializeParameter[p#0]   : &:r0_4
-#-----|     r0_6(Base *)         = CopyValue                  : r0_3
-#-----|     r0_7(glval<String>)  = FieldAddress[base_s]       : r0_6
-#  745|     r0_8(glval<unknown>) = FunctionAddress[operator=] : 
-#-----|     r0_9(glval<Base &>)  = VariableAddress[p#0]       : 
-#-----|     r0_10(Base &)        = Load                       : &:r0_9, ~mu0_2
-#-----|     r0_11(glval<String>) = FieldAddress[base_s]       : r0_10
-#  745|     r0_12(String &)      = Call                       : func:r0_8, this:r0_7, 0:r0_11
-#  745|     mu0_13(unknown)      = ^CallSideEffect            : ~mu0_2
-#-----|     r0_14(glval<Base &>) = VariableAddress[#return]   : 
-#-----|     r0_15(Base *)        = CopyValue                  : r0_3
-#-----|     mu0_16(Base &)       = Store                      : &:r0_14, r0_15
-#  745|     r0_17(glval<Base &>) = VariableAddress[#return]   : 
-#  745|     v0_18(void)          = ReturnValue                : &:r0_17, ~mu0_2
-#  745|     v0_19(void)          = UnmodeledUse               : mu*
-#  745|     v0_20(void)          = ExitFunction               : 
+#  745|     v0_0(void)           = EnterFunction                    : 
+#  745|     mu0_1(unknown)       = AliasedDefinition                : 
+#  745|     mu0_2(unknown)       = UnmodeledDefinition              : 
+#  745|     r0_3(glval<Base>)    = InitializeThis                   : 
+#-----|     r0_4(glval<Base &>)  = VariableAddress[p#0]             : 
+#-----|     mu0_5(Base &)        = InitializeParameter[p#0]         : &:r0_4
+#-----|     r0_6(Base *)         = CopyValue                        : r0_3
+#-----|     r0_7(glval<String>)  = FieldAddress[base_s]             : r0_6
+#  745|     r0_8(glval<unknown>) = FunctionAddress[operator=]       : 
+#-----|     r0_9(glval<Base &>)  = VariableAddress[p#0]             : 
+#-----|     r0_10(Base &)        = Load                             : &:r0_9, ~mu0_2
+#-----|     r0_11(glval<String>) = FieldAddress[base_s]             : r0_10
+#  745|     r0_12(String &)      = Call                             : func:r0_8, this:r0_7, 0:r0_11
+#  745|     mu0_13(unknown)      = ^CallSideEffect                  : ~mu0_2
+#-----|     v0_14(void)          = ^IndirectReadSideEffect          : &:r0_7, ~mu0_2
+#-----|     v0_15(void)          = ^IndirectReadSideEffect[p#0]     : &:r0_11, ~mu0_2
+#-----|     r0_16(String)        = ^IndirectMayWriteSideEffect      : &:r0_7, ~mu0_2
+#-----|     mu0_17(String)       = ^IndirectMayWriteSideEffect[p#0] : &:r0_11, ~mu0_2
```

Comment:
Can you explain the difference between the side-effect instructions with and without a parameter name?

Reply:
Side effect instructions associated with either the qualifier or a varargs argument don't have an associated parameter.

Reply:
I think even `this` and varargs need something to identify them on these instructions. I think we'd discussed the idea of using an `ArgumentOperandTag` here, since it handles all of the above?

Reply:
We've discussed a few different things. I'd like to keep `Parameter`s in the interface to make connecting caller side effects to callee initializations and returns. That would be better implemented via a separate `newtype` with branches `TThis`, `TParameter(Parameter)`, and `TVarArg(int)`.

Reply:
I jsut realized keeping `Parameter` in the interface would interact badly with virtual calls. I'd still prefer a separate `newtype`. Should we discuss it tomorrow morning?

Reply:
This is the same `ArgumentId` we're discussing below, right?

Reply:
Yeah. I assume it would create a semantic conflict with #1600 because of the pyrameterized modules PR check